Soap/Client/Client.php

```diff
@@ -0,0 +1,315 @@
+<?php
+/*
+* This file is part of the CleverAge/ProcessBundle package.
+*
+* Copyright (C) 2017-2018 Clever-Age
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*/
+
+namespace CleverAge\ProcessBundle\Soap\Client;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class Client
+ *
+ * @author Madeline Veyrenc <mveyrenc@clever-age.com>
+ */
+class Client implements ClientInterface
+{
+    /**
+     * @var
+     */
+    private $code;
+
+    /**
+     * @var string
+     */
+    private $wsdl;
+
+    /**
+     * @var array
+     */
+    private $options = [];
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var \SoapClient
+     */
+    private $soapClient;
+
+    /**
+     * @var string
+     */
+    private $lastRequest;
+
+    /**
+     * @var string
+     */
+    private $lastRequestHeaders;
+
+    /**
+     * @var string
+     */
+    private $lastResponse;
+
+    /**
+     * @var string
+     */
+    private $lastResponseHeaders;
+
+    /**
+     * Client constructor.
+     *
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
```

Comment:
If the logger is the only property that is always not null, all the other properties should be nullable